relay_messages.go

```diff
@@ -118,6 +119,8 @@ func newLazyCallReq(f *Frame) lazyCallReq {
 			cr.caller = val
 		} else if bytes.Equal(key, _routingDelegateKeyBytes) {
 			cr.delegate = val
+		} else if bytes.Equal(key, _routingKeyKeyBytes) {
```

Comment:
Thanks for updating this as well. Can you also add some test cases for this in `relay_messages_test.go`